src/resolvers.ts

```diff
@@ -22,17 +22,34 @@ export const resolveConfigPath = async (path: Arrayable<string>) => (
  * @param srcDir
  * @returns array of resolved content globs
  */
-export const resolveContentPaths = (srcDir: string) => ([
-  `${srcDir}/components/**/*.{vue,js,ts}`,
-  `${srcDir}/layouts/**/*.vue`,
-  `${srcDir}/pages/**/*.vue`,
-  `${srcDir}/composables/**/*.{js,ts}`,
-  `${srcDir}/plugins/**/*.{js,ts}`,
-  `${srcDir}/utils/**/*.{js,ts}`,
-  `${srcDir}/{App,app}.{js,ts,vue}`,
-  `${srcDir}/{Error,error}.{js,ts,vue}`,
-  `${srcDir}/app.config.{js,ts}`
-])
+export const resolveContentPaths = (srcDir: string, nuxt = useNuxt()) => {
+  const r = (p: string) => p.startsWith(srcDir) ? p : `${srcDir}/${p}`
+  const extensionFormat = (s: string[]) => s.length > 1 ? `.{${s.join(',')}}` : `.${s.join('') || 'vue'}`
+
+  const defaultExtensions = extensionFormat(['js', 'ts', 'mjs'])
+  const sfcExtensions = extensionFormat(nuxt.options.extensions.map(e => e.replace(/^\.*/, '')))
+
+  return [
+    ...(() => {
+      if (nuxt.options.components) {
+        return (Array.isArray(nuxt.options.components) ? nuxt.options.components : typeof nuxt.options.components === 'boolean' ? ['components'] : nuxt.options.components.dirs).map(d => `${resolveAlias(typeof d === 'string' ? d : d.path)}/**/*${sfcExtensions}`)
+      }
+      return []
+    })(),
+
+    r(`${nuxt.options.dir.layouts}/**/*${sfcExtensions}`),
+    ...(nuxt.options.pages ? [r(`${nuxt.options.dir.pages}/**/*${sfcExtensions}`)] : []),
+
+    r(`${nuxt.options.dir.plugins}/**/*${defaultExtensions}`),
+    r(`${nuxt.options.dir.modules}/**/*${defaultExtensions}`),
+    ...nuxt.options.modulesDir.map(m => r(`${m}/**/*${defaultExtensions}`)),
+    ...(nuxt.options.imports.dirs || []).map(d => r(`${d}/**/*${defaultExtensions}`)),
```

Comment:
@maddysrc we can continue the import dirs conversation. I'd also add that the link you shared is for the imports module, so I'd think its safe to assume that import.dirs will include composables and utils (as added by the module)

Reply:
The module never actually modifies the config, playground seems to confirm that.